packages/dds/merge-tree/src/segmentPropertiesManager.ts

```diff
@@ -3,13 +3,16 @@
  * Licensed under the MIT License.
  */
 
-/* eslint-disable @typescript-eslint/no-non-null-assertion */
-
 import { assert } from "@fluidframework/core-utils/internal";
 
+import { DoublyLinkedList, iterateListValues } from "./collections/index.js";
```

Comment:
this file contains most of the important changes, and is essentially a rewrite. i recommend reviewing in split view, rather than inline diff, as the changes are not incremental, so inline diffs will not make very much sense.

Reply:
I've created a more focused PR just for these changes that may be easier to review: #23059